api/src/com/cloud/naming/ResourceNamingPolicyManager.java

```diff
@@ -0,0 +1,32 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.naming;
+
+import com.cloud.utils.component.Manager;
+
+public interface ResourceNamingPolicyManager extends Manager {
+
+
+    /**
+     * Finds a resource naming policy among the registered ones by its class
+     * @param policyClass
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    public <T extends ResourceNamingPolicy> T getPolicy(Class<T> policyClass);
```

Comment:
I'm all for simplifying the code as much as possible. CloudStack has reinvented a lot of wheels since its creation (the entire DAO being the biggest offender in my opinion).

If we can change these to Spring singletons and inject them across all relevant modules and maintain the ability to change them, then I am fine with the idea. Cross-module injection is accomplished by using the extension registries and referencing their "getRegistered" property in Spring expression language in the XML files.